frontend/public/actions/dashboards.ts

```diff
@@ -42,43 +42,57 @@ const dashboardsActions = {
   setError,
 };
 
-const fetchPeriodically: FetchPeriodically = async (dispatch, type, key, url, getState, fetch) => {
+const fetchPeriodically: FetchPeriodically = async (
+  dispatch,
+  type,
+  key,
+  getURL,
+  getState,
+  fetch,
+) => {
   if (!isWatchActive(getState().dashboards, type, key)) {
     return;
   }
   try {
     dispatch(updateWatchInFlight(type, key, true));
-    const data = await fetch(url);
+    const data = await fetch(getURL());
     dispatch(setData(type, key, data));
   } catch (error) {
     dispatch(setError(type, key, error));
   } finally {
     dispatch(updateWatchInFlight(type, key, false));
     const timeout = setTimeout(
-      () => fetchPeriodically(dispatch, type, key, url, getState, fetch),
+      () => fetchPeriodically(dispatch, type, key, getURL, getState, fetch),
       REFRESH_TIMEOUT,
     );
     dispatch(updateWatchTimeout(type, key, timeout));
   }
 };
 
-export const watchPrometheusQuery: WatchPrometheusQueryAction = (query, namespace) => (
+export const watchPrometheusQuery: WatchPrometheusQueryAction = (query, namespace, timespan) => (
   dispatch,
   getState,
 ) => {
-  const isActive = isWatchActive(getState().dashboards, RESULTS_TYPE.PROMETHEUS, query);
-  dispatch(activateWatch(RESULTS_TYPE.PROMETHEUS, query));
+  const queryKey = timespan ? query + timespan : query;
```

Comment:
Although unlikely, `queryKey` could clash since a query can end with a number. For example
- `METRIC > 160000` (no timespan) and `METRIC > 1` + `60000`, or
- `node_load15` + `400000` and `node_load1` + `5400000`

An easy fix would be to use a separator character that cannot possibly appear in a query.

Reply:
The character `@` should be a safe choice. The only way that could appear in a query would be if the query had a comment, but we know we don't have any comments in this case.

Reply:
thanks for the suggestion! Im adding `@` between query and timestamp now